apps/astarte_appengine_api/lib/astarte_appengine_api/device/device.ex

```diff
@@ -266,37 +277,17 @@ defmodule Astarte.AppEngine.API.Device do
         opts
       )
 
-      wrapped_value = wrap_to_bson_struct(mapping.value_type, value)
-
-      case interface_descriptor.type do
-        :properties ->
-          DataTransmitter.set_property(
-            realm_name,
-            device_id,
-            interface_descriptor.name,
-            path,
-            wrapped_value
-          )
-
-        :datastream ->
-          Queries.insert_path_into_db(
-            client,
-            device_id,
-            interface_descriptor,
-            endpoint_id,
-            path,
-            timestamp_micro,
-            div(timestamp_micro, 1000),
-            opts
-          )
-
-          DataTransmitter.push_datastream(
-            realm_name,
-            device_id,
-            interface_descriptor.name,
-            path,
-            wrapped_value
-          )
+      if interface_descriptor.type == :datastream do
+        Queries.insert_path_into_db(
```

Comment:
I'm not sure if we should do`insert_path_into_db` before `ensure_publish` or viceversa.